File: src/main/java/dk/cngroup/wishlist/entity/Wishlist.java
package dk.cngroup.wishlist.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Wishlist extends AuditableEntity {
    @JsonBackReference
    @ManyToOne
    private Client client;

    @ManyToMany(cascade = CascadeType.PERSIST)
    @OrderColumn
    private List<Product> products = new ArrayList<>();

    public Wishlist(List<Product> products) {
        this.products = products;
    }
}

File: src/main/kotlin/dk/cngroup/wishlist/entity/Wishlist.kt
package dk.cngroup.wishlist.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import jakarta.persistence.*
import org.springframework.data.jpa.repository.JpaRepository

@Entity
class Wishlist(
    @JsonBackReference
    @ManyToOne
    var client: Client? = null,

    @OrderColumn
    @ManyToMany(cascade = [CascadeType.PERSIST])
    var products: MutableList<Product> = arrayListOf()
) : AuditableEntity()

interface WishlistRepository : JpaRepository<Wishlist, Long>
